src/runtime-rs/crates/hypervisor/src/qemu/inner.rs

```diff
@@ -379,7 +379,17 @@ impl QemuInner {
 
     pub(crate) async fn capabilities(&self) -> Result<Capabilities> {
         let mut caps = Capabilities::default();
-        caps.set(CapabilityBits::FsSharingSupport);
+
+        // Confidential Guest doesn't permit virtio-fs.
+        let flags = if self.hypervisor_config().security_info.confidential_guest {
+            CapabilityBits::BlockDeviceSupport | CapabilityBits::BlockDeviceHotplugSupport
+        } else {
+            CapabilityBits::BlockDeviceSupport
+                | CapabilityBits::BlockDeviceHotplugSupport
+                | CapabilityBits::FsSharingSupport
```

Comment:
Not related to this patch, but when `shared_fs = none` is set, we should also make suyre that FsSharingSupport is disabled.

Reply:
No, no need. Just remember to also adjust the other commit after this one gets merged.
Sounds fair?

Reply:
Yeah, sound good to me. Thx